source/common/config/utility.h

```diff
@@ -209,13 +211,47 @@ class Utility {
     return *factory;
   }
 
+  template <class Factory> static Factory& getAndCheckFactoryByType(absl::string_view type) {
+    if (type.empty()) {
+      throw EnvoyException("Provided type for static registration lookup was empty.");
+    }
+    Factory* factory = Registry::FactoryRegistry<Factory>::getFactoryByType(type);
+    if (factory == nullptr) {
+      throw EnvoyException(
+          fmt::format("Didn't find a registered implementation for type: '{}'", type));
+    }
+    return *factory;
+  }
+
   /**
    * Get a Factory from the registry with error checking to ensure the name and the factory are
    * valid.
    * @param message proto that contains fields 'name' and 'typed_config'.
    */
   template <class Factory, class ProtoMessage>
   static Factory& getAndCheckFactory(const ProtoMessage& message) {
+    const ProtobufWkt::Any& typed_config = message.typed_config();
+    static const std::string& struct_type =
```

Comment:
Why can't `Struct` be treated the same way as any double registered type URL?

Reply:
There are two cases: 
- case 1: Struct inside TypedStruct inside typed_config -> handled just like a double registered type URL;
- case 2: Struct inside typed_config -> ignored, treated as if config field was set instead of typed_config.

Both cases are abnormal IMHO. Do you want to make case 2 behave like case 1?

Reply:
I think we might want that, to make it easier for folks to move to `TypedStruct` even without fixing their config uniqueness issues.

Reply:
Ok, made it fall back to `name` if a factory cannot be found because either it is untyped, or double-registered for a type.